tim/ddr/DDR_TOPOLOGY_6.txt

```diff
@@ -18,5 +18,5 @@ ddr_speedbin_index=11
 #16BIT
 ddr_bus_width_index=2
 
-#16Gbits (2GB)
-ddr_mem_size_index=5
```

Comment:
DDR_TOPOLOGY_5.txt describes the 8gbits layout. Your patch is actually making DDR_TOPOLOGY_5 == DDR_TOPOLOGY_6. I do not understand the reason for doing this.

Reply:
From https://github.com/ARM-software/arm-trusted-firmware/blob/master/docs/plat/marvell/armada/build.rst:
DDR4 1CS (5): EspressoBin V7 (1GB)
DDR4 2CS (6): EspressoBin V7 (2GB)

The way I read it:
topo 6 is 2 *1GB, just like topo 2 is 2 * 512MB

Reply:
So it sounds like a new topology that uses different DDR chips (not 2x K4A8G165WB-BCRC as it mentioned in the comment).
Does this new topology match the description in DDR_TOPOLOGY_3.txt? If not, just create a new topology file describing this new configuration. The GlobalScale tends to replace board components in their production line from time to time (probably for the cost reduction). These changes should not invalidate boards that are already in the field.

Reply:
It's not a new topology. 2*1GB for a total of 2GB is correct.

Also see topo 0 vs. topo2:
```
diff -u tim/ddr/DDR_TOPOLOGY_0.txt tim/ddr/DDR_TOPOLOGY_2.txt 
--- tim/ddr/DDR_TOPOLOGY_0.txt	2020-08-28 19:49:59.253834169 +0200
+++ tim/ddr/DDR_TOPOLOGY_2.txt	2020-08-28 19:49:59.253834169 +0200
@@ -3,15 +3,14 @@
 # each item is one parameter for DDR topology, which will be parsed by
 # DDR tool and generate DDR register list accordingly.
 # Supported Marvell boards:
-#       DDR3 1CS : DB-88F3720-DDR3-Modular (512MB);
-#                  EspressoBIN (512MB)
+#       DDR3 2CS : EspressoBIN (1GB)
 #
 
 #DDR3
 ddr_type=0
 
-#1CS
-ddr_cs_mask=1
+#2CS
+ddr_cs_mask=3
 
 #DDR3_1600K
 ddr_speedbin_index=12
```

both have `ddr_mem_size_index=3` (512MB). It's per chip, 2CS doubles it.

By that logic topo 6 needs to match topo 5, just with CS2 instead of CS1. Which it doesn't, currently it reads 2GB*2CS=4GB, which never existed and was always wrong. The patch just corrects this.

Reply:
For Espressobin v7 DDR4 2GB it is 2x K4A8G165WB-BCRC. It should be a mistake in the commit 845f670be9cbed3d3561a8d20c99f23e1b866daf, the DDR_TOPOLOGY_5.txt and DDR_TOPOLOGY_6.txt should have the same configuration except ddr_cs_mask. Current DDR_TOPOLOGY_6.txt describes a DDR4 2CS 4GB configuration. Maybe DDR_TOPOLOGY_3.txt also has incorrect configuration, since 2x 1GB is not equal to 4GB, it doesn't brick the device, but makes DB-88F3720-DDR4-Modular has only 2GB of available memory.

Reply:
diff tim/ddr/DDR_TOPOLOGY_5.txt tim/ddr/DDR_TOPOLOGY_6.txt 
6c6
< #       DDR4 1CS : ESPERSSOBIN-DDR4 single K4A8G165WB-BCRC (1GB)
---
> #       DDR4 2CS : ESPERSSOBIN-DDR4 dual K4A8G165WB-BCRC (2GB)
12,13c12,13
< #1CS
< ddr_cs_mask=1
---
> #2CS
> ddr_cs_mask=3
21,22c21,22
< #8Gbits (1GB)
< ddr_mem_size_index=4
---
> #16Gbits (2GB)
> ddr_mem_size_index=5

Single chip (#5) - 1GB/8Gb and Dual chip (#6) - 2GB/16Gb

Reply:
https://github.com/MarvellEmbeddedProcessors/A3700-utils-marvell/blob/A3700_utils-armada-18.12/tim/ddr/porting_guide.txt

ddr_mem_size_index refers to single chip size, not total, for topo 5 and 6 they should not be different.

Reply:
I see. I will not be able to merge it to the requested branch - it is 2 years old and frozen. The updated one is https://github.com/MarvellEmbeddedProcessors/A3700-utils-marvell/tree/A3700_utils-armada-18.12-fixed